### PR TITLE
make payments retry count configurable

### PR DESCRIPTION
**Pull request by mdayican**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/BPS-1075

### Change description ###

make payments retry count configurable
set the timeout 3 as we want to try more than one.
and set retry count to 2,  
2 retry and 1 normal flow, total max will be 3 attempt .

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/domains/payments/PaymentsPublisher.java [hunk 1]
```diff
@@ -67,21 +72,21 @@ public void send(PaymentCommand cmd) {
...
             queueClient.send(message);
             LOG.info("Sent message to payments queue. ID: {}, Label: {}",
                     message.getMessageId(),
                     message.getLabel());
         } catch (Exception ex) {
-            if (retry) {
+            if (retryCount > 0) {
                 LOG.error(
-                        "Sending to payment queue got error, Message ID: {}. Will retry....",
+                        "Sent message to payments queue got error, Message ID: {}.Remaining Retry Count:{}, Retrying..",
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/domains/payments/PaymentsPublisher.java]: space after `.` and `:`

**Comment by mdayican** [comment]
> 4 tries will result in 20 secs wait whereas ccd timeout is 10 secs

made the timeout 3 and retry 2.
I thought in chat we want  3 retry.?

File: src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/domains/payments/PaymentsPublisher.java [hunk 2]
```diff
@@ -25,12 +26,16 @@
     private final QueueClient queueClient;
     private final ObjectMapper objectMapper;
 
+    private final int retryCount;
+
     public PaymentsPublisher(
         @Qualifier("payments") QueueClient queueClient,
-        ObjectMapper objectMapper
+        ObjectMapper objectMapper,
+        @Value("${azure.servicebus.payments.manual-retry-count}") int retryCount
```
Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/domains/payments/PaymentsPublisher.java]: Maybe we should remove `manual`, as those retries aren't manual

Comment (lgonczar) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/domains/payments/PaymentsPublisher.java]: how about `max-retries` or `max-retry-count`?

Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/domains/payments/PaymentsPublisher.java]: I want to distinguish automatic queue retry and our retry

**Pull request merged**